src/electos/ballotmaker/__version__.py

```diff
@@ -1 +1 @@
-__version__ = "0.1.0"
+__version__ = "0.1.1"
```

Comment:
Best practice for consideration:

- Use  [`importlib.metadata`](https://docs.python.org/3.8/library/importlib.metadata.html) to import the version from `pyproject.toml`, because it's `DRYer`. (Unlike `setup.py` or `setup.cfg` the pyproject doesn't allow loading code into it so  `__version__` has to be set from project file instead of the other way around.)
- Consider changing `__version__.py` to `__about__.py`. `importlib.metadata` allows using all of the metadata in the [Python Packaging Core Metadata Specification](https://packaging.python.org/en/latest/specifications/core-metadata/) can be added, so that can include authors, license and so on.

_source_: StackOverflow ["How to specify version in only one place when using pyproject.toml?"](https://stackoverflow.com/questions/67085041/how-to-specify-version-in-only-one-place-when-using-pyproject-toml)

Reply:
@ion-oset I've burned a few cycles on this; it turns out that using `importlib.metadata` is non-trivial. I think I may have to use `poetry build` to create the necessary files to read version and package name from `pyproject.toml` -- in fact, it seems like poetry uses `pyproject.toml` to **generate** the files (like, say, `setup.py`) that `importlib.metadata` reads.

More here, from an issue on the poetry repo: [How to get version from pyproject.toml from python app? · Issue #273 · python-poetry/poetry](https://github.com/python-poetry/poetry/issues/273#issuecomment-401983643)

In the meantime, to keep things simple, I've done the following:

* Consolidated all global strings and values into the `constants.py` file, including VERSION and PACKAGE_NAME.
* Updated  `pyproject.toml` by hand so these values are in sync.
* Removed the `__verison__.py` file from the package, and abandoned the idea of an `__about__.py` file for the time being.

## Moving forward

One possible solution is to create an `__about__.py` file that uses `importlib.metadata` to get the **version** and **package_name** from the poetry-generated environment. Then I could update `constants.py` to read the values from `__about__.py` instead of updating them by hand. This way, I don't have to rewrite code that's getting the values from `constants.py` but we can effectively read the values from `pyproject.toml`.s 

This summary is basically a restatement of what you're proposing above, as I understand it. But, I don't have the cycles to work on this right now -- is this something you're interested in? I'm going to create an issue with the description I just typed in. Let me know if you want to work on that.

In the meantime, I think that using `constants.py` to store global constants is suifficient for this PR.

Thoughts?

Reply:
I don't think this issue is high priority. I mentioned it as best practice, and we should get to it eventually but we don't need to resolve it right now. Don't block this PR on it.

I added it to the project tracker as "Specify how project metadata works for Python" and starred it so it goes in my issues. 

 > In the meantime, I think that using `constants.py` to store global constants is sufficient for this PR.
 
 Agreed 👍🏼 . The only qualification I'd make is that `__about__.py` and `constants.py` have different purposes:
 
- `__about__.py` (by whatever name) provide _external_ visibility for the package metadata to code that needs to know that sort of thing (say build scripts).
- `constants.py` is _internal_ to the package. It's thus not limited to public data.

Reply:
@ion-oset your distinction between `__about__.py` and `constants.py` makes sense to me, so I'll consider this conversation resolved.

Please be sure to indicate that your review is complete after you've looked at everything, or add new comments if you think there's anything else that needs my attention.

Reply:
This item is now an issue, so I'll mark this resolved.